app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,64 @@
+class OrdersController < ApplicationController
+  def index
+    @orders = Order.for_user(current_user)
+  end
+
+  def show
+    @order = Order.find(params[:id])
+  end
+
+  def no_cart
+    flash[:alert] = 'you have not put anything in the cart'
+    redirect_to root_path
+  end
+
+  def create
+    if current_user.present?
+      order = find_order
+      if order.blank?
+        params[:alert] = "This Order is not belongs to you"
+        redirect_to :back
+        return
+      end
+
+      if !params[:order_detail][:quantity].to_s.match(/\A[+-]?\d+?(\.\d+)?\Z/) || params[:order_detail][:quantity].to_i < 0
+        flash[:alert] = 'quantity must be number and greater than 0'
+        redirect_to :back
+        return
```

Comment:
when you do `redirect_to` and `render`, it's still run the other code, if it found another `redirect_to`, it will raise error. Furthermore since it's still running other code, `Order` record will be created, and it's detail too